### PR TITLE
feat: implement the delete handler and its logic in the tree service 

### LINKED ISSUE #82: Implement delete tree in tree service

### PR DESCRIPTION
**Pull request by Mohammad96Assaf**
Close #82

### CODE REVIEW

File: internal/server/http/handler/v1/tree/handler.go [hunk 1]
```diff
@@ -142,10 +142,18 @@ func UpdateTree(_ service.TreeService) fiber.Handler {
 // @Failure		500	{object}	HTTPError
 // @Router			/v1/tree [delete]
 // @Param			Authorization	header	string	true	"Insert your access token"	default(Bearer <Add access token here>)
```
Comment (choffmann) [inline internal/server/http/handler/v1/tree/handler.go:144]: Please add the id to swagger

```suggestion
// @Router			/v1/tree/{tree_id} [delete]
// @Param			treecluster_id	query	string	false	"Tree Cluster ID"
// @Param			Authorization	header	string	true	"Insert your access token"	default(Bearer <Add access token here>)
```

**Review by choffmann — CHANGES_REQUESTED** [review]
:1st_place_medal:

File: internal/server/http/handler/v1/tree/handler.go (hunk 1 above)
Comment (Mohammad96Assaf) [reply internal/server/http/handler/v1/tree/handler.go]: Why treecluster_id, you mean tree_id ?

File: internal/server/http/handler/v1/tree/handler.go [hunk 2]
```diff
@@ -140,12 +140,21 @@ func UpdateTree(_ service.TreeService) fiber.Handler {
 // @Failure		403	{object}	HTTPError
 // @Failure		404	{object}	HTTPError
 // @Failure		500	{object}	HTTPError
-// @Router			/v1/tree [delete]
+// @Router			/v1/tree/{tree_id} [delete]
+// @Param			tree_id			query	string	false	"Tree ID"
```
Comment (choffmann) [inline internal/server/http/handler/v1/tree/handler.go:144]: It's not a query parameter, it's a path parameter.

```suggestion
// @Param			tree_id			path	string	false	"Tree ID"
```

**Review by choffmann — APPROVED** [review]
Please change query to path, then you can merge the pr

**Pull request merged**